Source/WebCore/editing/TextManipulationController.cpp

```diff
@@ -717,8 +718,12 @@ auto TextManipulationController::completeManipulation(const Vector<WebCore::Text
         std::exchange(itemData, itemDataIterator->value);
         m_items.remove(itemDataIterator);
 
-        if (auto failureOrNullopt = replace(itemData, itemToComplete.tokens, containersWithoutVisualOverflowBeforeReplacement))
+        if (auto failureOrNullopt = replace(itemData, itemToComplete.tokens, containersWithoutVisualOverflowBeforeReplacement)) {
             failures.append(ManipulationFailure { *frameID, itemID, i, *failureOrNullopt });
+            continue;
+        }
+
+        succeededIndexes.append(i);
```

Comment:
Instead of appending a full integer to store the index of success, why don't we create BitVector of length matching the length of completionItems? Then you can just check BitVector::get(index) to check whether the given index has succeeded or not.

Reply:
`BitVector` is not supported in IPC message yet. I can use `BitVector` here but will need to convert it to `Vector` for sending reply to UI process.

Reply:
Oh I see. That's unfortunate.